src/controllers/dynakube/dynakube_controller.go

```diff
@@ -174,183 +170,172 @@ func (controller *DynakubeController) reconcileIstio(dynakube *dynatracev1beta1.
 	}
 	return err
 }
 
-func updatePhaseIfChanged(instance *dynatracev1beta1.DynaKube, newPhase dynatracev1beta1.DynaKubePhaseType) bool {
-	if instance.Status.Phase == newPhase {
-		return false
-	}
-	instance.Status.Phase = newPhase
-	return true
+func (controller *DynakubeController) removeOneAgentDaemonSet(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) error {
+	ds := appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: dynakube.OneAgentDaemonsetName(), Namespace: dynakube.Namespace}}
+	return kubeobjects.Delete(ctx, controller.client, &ds)
 }
 
-func (controller *DynakubeController) removeOneAgentDaemonSet(dkState *status.DynakubeState) {
-	ds := appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: dkState.Instance.OneAgentDaemonsetName(), Namespace: dkState.Instance.Namespace}}
-	if err := kubeobjects.Delete(context.TODO(), controller.client, &ds); dkState.Error(err) {
-		return
-	}
-}
+func (controller *DynakubeController) reconcileActiveGate(ctx context.Context, dynakube *dynatracev1beta1.DynaKube, dtc dtclient.Client) error {
+	reconciler := activegate.NewReconciler(ctx, controller.client, controller.apiReader, controller.scheme, dynakube, dtc)
+	_, err := reconciler.Reconcile()
 
-func (controller *DynakubeController) reconcileActiveGate(ctx context.Context, dynakubeState *status.DynakubeState, dtc dtclient.Client) error {
-	reconciler := activegate.NewReconciler(ctx, controller.client, controller.apiReader, controller.scheme, dynakubeState.Instance, dtc)
-
-	upd, err := reconciler.Reconcile()
 	if err != nil {
 		return errors.WithMessage(err, "failed to reconcile ActiveGate")
 	}
-
-	dynakubeState.Update(upd, "ActiveGate reconciled")
-	controller.startApiMonitoring(dynakubeState, dtc)
+	controller.setupAutomaticApiMonitoring(dynakube, dtc)
 
 	return nil
 }
 
-func (controller *DynakubeController) startApiMonitoring(dynakubeState *status.DynakubeState, dtc dtclient.Client) {
-	if dynakubeState.Instance.Status.KubeSystemUUID != "" &&
-		dynakubeState.Instance.FeatureAutomaticKubernetesApiMonitoring() &&
-		dynakubeState.Instance.IsKubernetesMonitoringActiveGateEnabled() {
+func (controller *DynakubeController) setupAutomaticApiMonitoring(dynakube *dynatracev1beta1.DynaKube, dtc dtclient.Client) {
+	if dynakube.Status.KubeSystemUUID != "" &&
+		dynakube.FeatureAutomaticKubernetesApiMonitoring() &&
+		dynakube.IsKubernetesMonitoringActiveGateEnabled() {
 
-		clusterLabel := dynakubeState.Instance.FeatureAutomaticKubernetesApiMonitoringClusterName()
+		clusterLabel := dynakube.FeatureAutomaticKubernetesApiMonitoringClusterName()
 		if clusterLabel == "" {
-			clusterLabel = dynakubeState.Instance.Name
+			clusterLabel = dynakube.Name
 		}
 
-		err := apimonitoring.NewReconciler(dtc, clusterLabel, dynakubeState.Instance.Status.KubeSystemUUID).
+		err := apimonitoring.NewReconciler(dtc, clusterLabel, dynakube.Status.KubeSystemUUID).
 			Reconcile()
 		if err != nil {
 			log.Error(err, "could not create setting")
```

Comment:
replace `log.Error`. do we just ignore this error here?

Reply:
we ignore the error as far as I remember so that is why we log the whole error